lib/infrastructure/participation/participation_repository.dart

```diff
@@ -121,17 +120,66 @@ class ParticipationRepository implements IParticipationRepository {
 
       final participations = itemsJson
           .map(
-            (item) => ParticipationDto.fromJson(item as Map<String, dynamic>),
+            (item) => ParticipationDto.fromJson(item as Map<String, dynamic>)
+                .toDomain(),
           )
           .toList();
 
       final pageInfo = PageInfoDto.fromJson(pageInfoJson);
 
+      // To display user as first participant,
+      // Get user participation for first page
+      if (pageNumber == 1) {
+        int index = participations.indexWhere(
+          (p) => p.userId == authRepository.currentUser.id,
+        );
+
+        // If user already among participants
+        // skip fetching user participation
+        if (index > -1) {
+          // If user participation is in list,
+          // and not first, swap with first user
+          if (index > 0) {
+            final temp = participations[index];
+            participations[index] = participations.first;
+            participations[0] = temp;
+          }
+
+          return right(
+            PaginatedParticipations(
+              participations: participations,
+              pageInfo: pageInfo.toDomain(),
+            ),
+          );
+        }
```

Comment:
What if the user isn't on the first page of participations, we should additionally check if it's on any of the latter pages and remove the user I guess.